external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+if(${ENABLE_gauxc})
+    if((NOT "${gauxc_DIR}" STREQUAL "") AND "${IntegratorXX_DIR}" STREQUAL "")
+       set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+    endif()
+
+    find_package(ExchCXX CONFIG PATHS ${gauxc_DIR}/../ExchCXX) 
+    find_package(gauxc 0.0.1 CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== check if GPU-enabled GauXC is found, if GPU is specified ==#
+        get_property(_gpu TARGET gauxc::gauxc PROPERTY GAUXC_HAS_CUDA)
+        
+        if(${gauxc_ENABLE_GPU} AND (NOT ${_gpu}))
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned on, but selected GauXC install is not built for GPU support!")
+        elseif((NOT ${gauxc_ENABLE_GPU}) AND ${_gpu})
+           message(FATAL_ERROR "gauxc_ENABLE_GPU turned off, but selected GauXC install is built for GPU support!")
+        endif()
+
+        #== normal print-out and library addition ==#
+        get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION_RELWITHDEBINFO)
+        #get_property(_loc TARGET gauxc::gauxc PROPERTY IMPORTED_LOCATION)
+        message(STATUS "${Cyan}Found gauxc ${ColourReset}: ${_loc} (found version ${gauxc_VERSION})")
+        add_library(gauxc_external INTERFACE)  # dummy
+
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(VERBOSE "Suitable gauxc located externally as user insists")
+        endif()
+
+    else()
+        if(${CMAKE_INSIST_FIND_PACKAGE_gauxc})
+            message(FATAL_ERROR "Suitable gauxc could not be externally located as user insists")
+        endif()
+
+        message(STATUS "Suitable gauxc could not be located, ${Magenta}Building gauxc${ColourReset} instead.")
+
+        include(ExternalProject)
+        ExternalProject_Add(gauxc_external
+            URL https://github.com/wavefunction91/GauXC/archive/issue_56_58/reduce_shell_pair_mem.tar.gz 
+            UPDATE_COMMAND ""
+            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${STAGED_INSTALL_PREFIX}
+                   -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                   -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+                   -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+                   -DCMAKE_INSTALL_LIBDIR=${CMAKE_INSTALL_LIBDIR}
+                   -DCMAKE_INSTALL_INCLUDEDIR=${CMAKE_INSTALL_INCLUDEDIR}
+                   -DCMAKE_OSX_DEPLOYMENT_TARGET=${CMAKE_OSX_DEPLOYMENT_TARGET}
+                   -DCMAKE_OSX_SYSROOT=${CMAKE_OSX_SYSROOT}
+                   -DGAUXC_ENABLE_CUDA=${gauxc_ENABLE_GPU}
+                   -DCMAKE_CUDA_ARCHITECTURES=70
```

Comment:
This should not be hardcoded, it should be manually specified and propagated. GauXC will fail if it's not passed

Reply:
I've started adding this functionality, by making `CMAKE_CUDA_ARCHITECTURES` a user-facing option in Psi4. I have yet to fully test a compilation with it, however.

Reply:
I'd be interested to see how you do this and to get @loriab's input here. Technically, `CMAKE_CUDA_ARCHITECTURES` is meant to be a build-wide constant. https://cmake.org/cmake/help/latest/variable/CMAKE_CUDA_ARCHITECTURES.html

To override it would be akin to overriding `CMAKE_<LANG>_FLAGS`

It can be made target specific
https://cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html

But I've never seen that done, and I suspect it would be very error prone. 

If someone is going to build Psi4 w/GPU support, they're likely going to want to build the full project to target a specific microarchitecture, so straying from the standard variables might be a bit confusing.

Reply:
I think this comment right here is the final issue to address from the build setup, at least among those presented in this PR review. Right now, I handle this by adding `CMAKE_CUDA_ARCHITECTURES` to the list of default options in the Psi4 build list, in line 159 of `CMakeLists.txt`, as follows:

`option_with_default(CMAKE_CUDA_ARCHITECTURES "CUDA CCs to build GPU addons for" 70)`

Which gives a default value of 70 for the CC to use. @loriab, what do you think?

Reply:
I suppose we should probably reignite this discussion. @loriab, any thoughts as to the best way to handle propogating a Psi4-input `CMAKE_CUDA_ARCHITECTURES` value to an internally-built GauXC instance? @wavefunction91 brings up some good points on this front regarding how I have it set up right now.

Reply:
@davpoolechem I would not default that, or if you do default it, I'd make it the absolute minimum required (60 for GauXC). 70 is V100, not everyone is going to have compatible hardware and defaulting will just lead to silent runtime errors. GauXC will check compatibility of `CMAKE_CUDA_ARCHITECTURES` FWIW, so setting it to some small value will be fine and lead to configure errors forcing the user to set it to something sane.

Reply:
Do you think adding something like

```
if(${gauxc_ENABLE_GPU} AND (NOT DEFINED ${CMAKE_CUDA_ARCHITECTURES}))
    message(FATAL_ERROR "Internal GPU-enabled GauXC construction specified, but CMAKE_CUDA_ARCHITECTURES is not defined!")
endif()
```

alongside removing defaulting `CMAKE_CUDA_ARCHITECTURES` would be an acceptable solution? This gets around any need for setting a default by simply having configuration fail if the user tries to build an internal GPU-enabled GauXC instance without specifying `CMAKE_CUDA_ARCHITECTURES` explicitly. I'm also happy to use your solution of using some small default value if you prefer that approach instead.

Reply:
This would be a good solution

Reply:
All right, done!